File: punto4.c
#include <stdio.h>
#include <time.h>

int main() {
    long long int suma = 0;
    long long int n = 1000000000; // 1 billón

    // Iniciar el temporizador
    clock_t start = clock();

    // Calcular la suma de los primeros N números
    for (long long int i = 1; i <= n; i++) {
        suma += i;
    }

    // Finalizar el temporizador
    clock_t end = clock();
    double time_spent = (double)(end - start) / CLOCKS_PER_SEC;

    printf("Suma: %lld\n", suma);
    printf("Tiempo de ejecución en C: %f segundos\n", time_spent);

    return 0;
}

File: punto4.py
import time

suma = 0
n = 1000000000  # 1 billón

# Iniciar el temporizador
start_time = time.time()

# Calcular la suma de los primeros N números
for i in range(1, n + 1):
    suma += i

# Finalizar el temporizador
end_time = time.time()
time_spent = end_time - start_time

print(f"Suma: {suma}")
print(f"Tiempo de ejecución en Python: {time_spent} segundos")

